javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,19 @@
+var $characters = document.querySelectorAll('span');
+
+document.addEventListener('keydown', logKey);
+
+var currentChar = 0;
+
+function logKey(event) {
+  if (event.key === $characters[currentChar].textContent) {
+    $characters[currentChar + 1].className = 'margin green black-underline';
+    $characters[currentChar].className = 'margin green';
+    currentChar++;
+  } else if (event.code === 'Space' && $characters[currentChar].getAttribute('space') === 'true') {
```

Comment:
We should talk about how you can change how you handle spaces, it might just require a reorganization of your HTML. I would start by putting all the `span` elements for each word on one line, with the space after it on that same line. If you do that, you can likely make the text content an actual space character. Let me know if it doesn't got he way you expect, I am happy to help!

Reply:
ohhhhhh
that would eliminate the need for arbitrary classes lol

Reply:
and attributes 😅